### ISSUE TITLE
unused_lifetimes false positive when lifetime is used on generic bounds

### ISSUE THREAD

**Issue by crumblingstatue** [Manishearth/rust-clippy#489]
```Rust
pub fn parse<'a, I: Iterator<Item=&'a str>>(_it: &mut I) {
    unimplemented!()
}

```

```
src/lib.rs:1:14: 1:16 warning: this lifetime isn't used in the function definition, #[warn(unused_lifetimes)] on by default
src/lib.rs:1 pub fn parse<'a, I: Iterator<Item=&'a str>>(_it: &mut I) {
                          ^~
src/lib.rs:1:14: 1:16 help: for further information visit https://github.com/Manishearth/rust-clippy/wiki#unused_lifetimes
```

The wiki entry for unused_lifetimes is also missing.

**Comment by llogiq** [comment]
I'm working on putting the wiki entries into the code and generating the wiki page from that.

**Comment by Manishearth** [comment]
This is pretty easy to fix, we just need to walk the type parameter bounds (but not lifetime parameter bounds) as well.

**Comment by llogiq** [comment]
Thanks for making us aware of the bug, @crumblingstatue :+1:

**Comment by crumblingstatue** [comment]
> Thanks for making us aware of the bug, @crumblingstatue

No problem, thanks for fixing it so quickly!
